src/core/symbology-ng/qgsgeometrygeneratorsymbollayerv2.cpp

```diff
@@ -186,17 +186,22 @@ bool QgsGeometryGeneratorSymbolLayerV2::isCompatibleWithSymbol( QgsSymbolV2* sym
   Q_UNUSED( symbol )
   return true;
 }
-
 void QgsGeometryGeneratorSymbolLayerV2::render( QgsSymbolV2RenderContext& context )
 {
-  QgsGeometry geom = mExpression->evaluate( &context.renderContext().expressionContext() ).value<QgsGeometry>();
   if ( context.feature() )
   {
-    QgsFeature f = *context.feature();
+    QgsExpressionContext& expressionContext = context.renderContext().expressionContext();
+
+    QgsFeature f = expressionContext.feature();
```

Comment:
The original `QgsFeature f = *context.feature();` would be faster, is there any particular reason for this change?